miden-crypto/src/merkle/smt/mod.rs

```diff
@@ -71,12 +71,17 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
     // PROVIDED METHODS
     // ---------------------------------------------------------------------------------------------
 
-    /// Returns a [MerklePath] to the specified key.
+    /// Returns a [SparseMerklePath] to the specified key.
     ///
     /// Mostly this is an implementation detail of [`Self::open()`].
-    fn get_path(&self, key: &Self::Key) -> MerklePath {
+    fn get_path(&self, key: &Self::Key) -> SparseMerklePath {
         let index = NodeIndex::from(Self::key_to_leaf_index(key));
-        index.proof_indices().map(|index| self.get_node_hash(index)).collect()
+
+        // SAFETY: this is guaranteed to have depth <= SMT_MAX_DEPTH
+        SparseMerklePath::from_sized_iter(
+            index.proof_indices().map(|index| self.get_node_hash(index)),
+        )
+        .expect("failed to convert to SparseMerklePath")
```

Comment:
nit: expect str should be phrased in terms of what is expected. i.e. what the safety comment above says - that iterator len is <= SMT_MAX_DEPTH.

side note: do we bother with safety comments when an expect statement is provided to the same effect? @bobbinth

Reply:
I think the `expect()` message is fine here (don't think we have any guidelines documented on this).

I also think the safety comment helps. I would maybe add to it that `depth <= SMT_MAX_DEPTH` is enforced by `NodeIndex` to be clear where this assumption is coming from.

But, now that I look at it, I don't think we strictly enforce in the `NodeIndex` that `depth` must be less than 64. It is kind of assumed because `value` is a `u64`, but from what I can tell, someone could still create a `NodeIndex` with `value = 0` and `depth = 256`. We should fix this - but doesn't have to be done in this PR (let's create an issue for this though).

Reply:
And a PR: https://github.com/0xMiden/crypto/pull/482